### ISSUE TITLE
When using binding, on enter key pressed I do not receive the actual input value

### ISSUE THREAD

**Issue by NikolayIT** [aspnet/AspNetCore#15635]
I have the following code:

    <input bind=@this.Value onkeypress="@KeyPress" />
    @functions {
        public string Value { get; set; }
        void KeyPress(UIKeyboardEventArgs e)
        {
            if (e.Key == "Enter")
            {
                Console.WriteLine(this.Value);
            }
        }
    }

When I write some text in the input field (for example `test1`) and click enter I get empty text written on the console. When I add some more text (like `test2`) and again click enter I get just `test1` (the previous value before the enter click) on the console:

![image](https://user-images.githubusercontent.com/3106986/45209844-a9e51000-b296-11e8-8e41-83c8c1e5c325.png)

**Comment by Webreaper** [comment]
So i just got this working. You may want to try the following:

    <input @bind-value="MyValue" @bind-value:event="oninput" @onkeyup="DoTheThing" />

and then your @Code:

    public string MyValue { get; set; }
    private void DoTheThing( KeyboardEventArgs eventArgs)
    {
        if (eventArgs.Key == "Enter" )        // fire on enter 
        {
            Console.WriteLine( "Done the thing: "+ MyValue);
        }        
    }

Basically, the oninput bind-value:event saves the value of the input to the variable as you type, and then the Enter trigger allows you to process it when the user clicks return.
